examples/threads.c

```diff
@@ -1,3 +1,27 @@
+/*
+Example of instantiating of the WebAssembly module and invoking its exported
+function in a separate thread.
+
+You can compile and run this example on Linux with:
+
+   cargo build --release -p wasmtime-c-api
+   cc examples/threads.c \
+       -I crates/c-api/include \
+       -I crates/c-api/wasm-c-api/include \
+       target/release/libwasmtime.a \
+       -lpthread -ldl -lm \
```

Comment:
nit: I believe you have to use `-pthread` instead of `-lpthread` to be portable across UNIXes.

Reply:
That was a copy and paste from serialize.c, which uses `-lpthread`, copied so cmake could be mentioned, I could change the rest to be consistent if you'd like